### PR TITLE
Add cropping for LocalImages

### PR DESCRIPTION
**Pull request by karinassuni**
## Summary

In re: #77 
Implements cropping for LocalImages. The cropping happens before resizing the rendition, so a cropped image can be subsequently rescaled.

## Detailed description

This PR adds cropping functionality to LocalImages. A new argument to Markdown image inlines, `crop=(left,top,right,bottom)`, will be processed to this end. To handle cropping "before" resizing logic so that they can happen separately and synergize, I calculate the size of the crop box and use it for the input size in the resizing calculation function `LocalImage.get_rendition_size`. Resizing based off of the dimensions of the original image wouldn't work of course since cropping gives us a new image with new dimensions. In `LocalImage._render`, [PIL.Image.Image.crop](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.crop) actually does the cropping for the generated image, and happens right before `PIL.Image.Image.resize`.

## Developer/user impact

The user now has the option to crop local images via a `crop=(l,t,r,b)` argument to image inlines in Markdown.

## Test plan

https://github.com/PlaidWeb/publ-site/pull/4 includes one example/test of resizing and cropping simultaneously in the "Image rendition tests" entry. One can test this functionality in an entry against other crop values—not included in the PR is the test `alt text} "test crop double the size of the original image dimensions")`, which actually just extends the image to double the size, but filling in the new size with a black background. I'm not sure what the desired functionality should be in this case of probable user error.

### CODE REVIEW

File: publ/image.py [hunk 1]
```diff
@@ -304,7 +310,7 @@ def _image(self):
         return image
 
     # pylint:disable=too-many-arguments
-    def _render(self, path, size, box, flatten, kwargs, out_args):
+    def _render(self, path, crop, size, box, flatten, kwargs, out_args):
```
Comment (fluffy-critter) [inline publ/image.py]: The `LocalImage.render` argument list is getting comically long and in a future refactor I just want to separate the output filename generation out and only pass `kwargs` in, but that's outside the scope of this specific change so I'm mostly just rambling. :)

**Comment by fluffy-critter** [comment]
Ah, heh, looking at the Pillow docs, `Image.crop` *used* to be lazy but now it happens immediately. Not that it even matters.

Oh, and regarding how to handle the user error of a crop rectangle exceeding the original bounds, I think just letting Pillow handle that is fine.

**Comment by fluffy-critter** [comment]
Actually I'll just accept this PR and make my preferred changes myself while I refactor things a bit anyway. Thanks for contributing!

**Comment by karinassuni** [comment]
Alrightie times two! Thank you for accepting my contribution!

**Pull request merged**